src/main.ts

```diff
@@ -135,7 +142,25 @@ export function generateFile(typeMap: TypeMap, fileDesc: FileDescriptorProto, pa
   }
 
   visitServices(fileDesc, sourceInfo, (serviceDesc, sInfo) => {
-    file = file.addInterface(generateService(typeMap, fileDesc, sInfo, serviceDesc, options));
+    file = file.addInterface(
+      options.nestJs
+        ? generateNestjsService(typeMap, fileDesc, sInfo, serviceDesc, options)
+        : generateService(typeMap, fileDesc, sInfo, serviceDesc, options)
+    );
+    if (options.nestJs) {
+      // generate nestjs grpc client interface
+      file = file.addInterface(generateNestjsServiceClient(typeMap, fileDesc, sInfo, serviceDesc, options));
+
+      // generate nestjs grpc service controller decorator
+      file = file.addFunction(generateNestjsGrpcServiceMethodsDecorator(serviceDesc, options));
+
+      let serviceConstName =  `${camelToSnake(serviceDesc.name)}_NAME`;
+      if(!serviceDesc.name.toLowerCase().endsWith('service')){
```

Comment:
Not quite following this? Would it hurt to just always export the const? Even if the user, should their service end with `service` doesn't technically need it, it's there for consistency?

Reply:
Ah I quite like that, @stephenh i think its explained here 

>  Note: Based on the `.proto` we'll generate a `const` for example `HERO_PACKAGE_NAME` and `HERO_SERVICE_NAME` this way your code breaks if you change your package or service name. (It's safer to have compiler errors than runtime errors)

Reply:
@stephenh This way we offer more consistency in the naming. If the user happened to have a service inside their `.proto` named: `hero` then it would end up as `HERO_NAME`. We could do it without forcing the `SERVICE` in here but this also makes me think. If we go for this approach we also should force the `service` in the `controller` and `client` interface name?

@stephenh, @iangregsondev let me know what you guys think is a better approach.

Reply:
Ah okay. 

> If we go for this approach we also should force the service in the
> controller and client interface name?

Eh, I could go either way. I think what you have now is fine (i.e. don't force `Service` into the interface names).

Most users probably do have `Service` as a suffix of their service names anyway? Or maybe they prefer something like `HeroRpc` (just making something up, but some other suffix that is "like service") so by assuming "oh you must want `Service` suffixed", we'd be doing the wrong thing.

Better to make less assumptions I think.

In that vein, I'm tempted to nudge the "does name end in service" check here out to be consistent, i.e. maybe they want would `HERO_RPC_NAME` :shrug: .

(I agree having `service Hero` is kind of odd, so I'm assuming most users won't do that.)

Reply:
Either `HeroRpc` or `HeroService` look fine to me but I prefer `HeroService` since it's a `service` inside the proto.

Also I think we should no force the user to much as you said.

So we can resolve this issue and leave it as it is now?